### PR TITLE
upgraded to the latest version of jars used in HMPPS

### PR DESCRIPTION
**Pull request by sacintha**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,10 +1,17 @@
 plugins {
-  id("uk.gov.justice.hmpps.gradle-spring-boot") version "4.8.4"
-  kotlin("plugin.spring") version "1.8.10"
-  kotlin("plugin.jpa") version "1.8.10"
+  id("uk.gov.justice.hmpps.gradle-spring-boot") version "5.4.1"
```
Comment (petergphillips) [inline build.gradle.kts:2]: latest is 5.5.1

File: build.gradle.kts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+  kotlin("plugin.spring") version "1.8.22"
```
Comment (petergphillips) [inline build.gradle.kts:3]: should really go to 1.9.10 here

File: build.gradle.kts [hunk 3]
```diff
@@ -37,28 +44,33 @@ tasks.named("check") {
...
+  implementation("com.microsoft.azure:applicationinsights-logging-logback:2.6.4")
 
   // GOVUK Notify:
-  implementation("uk.gov.service.notify:notifications-java-client:3.17.3-RELEASE")
-
+  implementation("io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:1.29.0")
   // Enable kotlin reflect
-  implementation("org.jetbrains.kotlin:kotlin-reflect:1.6.20")
-
+  runtimeOnly("org.jetbrains.kotlin:kotlin-reflect:1.6.0-M1")
```
Comment (petergphillips) [inline build.gradle.kts:60]: not really recommended to go to milestone versions

File: build.gradle.kts [hunk 4]
```diff
@@ -87,12 +101,12 @@ jacoco {
 
 kotlin {
   jvmToolchain {
-    this.languageVersion.set(JavaLanguageVersion.of("18"))
+    this.languageVersion.set(JavaLanguageVersion.of("19"))
```
Comment (petergphillips) [inline build.gradle.kts:104]: odd to go from 18 to 19, would suggest 20 instead

Comment (petergphillips) [reply build.gradle.kts:104]: if you're changing this then you also need to change your Dockerfile and .circleci/config.yaml

File: src/main/resources/db/migration/V1_1__start_PG.sql [hunk 5]
```diff
@@ -0,0 +1,61 @@
+create table IF NOT EXISTS ability_to_work_impact (offender_id varchar(255) not null, work_impact integer);
```
Comment (petergphillips) [inline src/main/resources/db/migration/V1_1__start_PG.sql:1]: would be better not to include these changes if you're upping the versions and do the flyway stuff in a different PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/integration/IntegrationTestBase.kt [hunk 6]
```diff
@@ -24,6 +25,8 @@ abstract class IntegrationTestBase internal constructor() {
 
   @Autowired
   lateinit var jwtAuthHelper: JwtAuthHelper
+
+  val mapper = jacksonObjectMapper()
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/integration/IntegrationTestBase.kt:29]: better to use the autowired object mapper instead

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/entity/WorkInterests.kt [hunk 7]
```diff
@@ -49,9 +47,6 @@ data class WorkInterests(
   @Schema(description = "This is the list of detailed interests of the inmate.", name = "particularJobInterests", required = true)
   var particularJobInterests: MutableSet<WorkInterestDetail>,
 
-  @OneToOne(mappedBy = "workInterests")
-  @JsonIgnore
-  var previousWork: PreviousWork?,
```
Comment (nathanrussell-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/entity/WorkInterests.kt:54]: @mattwills-moj-digital - this could be of interest / relevant to what you are currently doing ?

File: src/main/resources/db/migration/V1_1__start_PG.sql (hunk 5 above)
Comment (nathanrussell-moj-digital) [reply src/main/resources/db/migration/V1_1__start_PG.sql:1]: 💯 agreed;

Also, will this even work? (given that this repo with the existing flyway scripts has already been deployed). Can you delete previously run flyway scripts?

Even if technically you can (delete previously run flyway migrations) I would question the wisdom of doing that.
If you delete them then there is no (obvious) history of the database migration over time .... which is a big part of what tools like flyway bring to the party.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/entity/EducationAndQualification.kt [hunk 8]
```diff
@@ -52,13 +50,10 @@ data class EducationAndQualification(
   @Schema(description = "This is the additional which is peculiar to this inmate  .This field is mandatory when  \"additionalTraining\" has a Value set to \"OTHER\" ", name = "additionalTrainingOther", required = false)
   var additionalTrainingOther: String?,
 
-  @OneToOne(mappedBy = "qualificationsAndTraining")
-  @JsonIgnore
-  var profile: CIAGProfile?,
```
Comment (mattwills-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/entity/EducationAndQualification.kt:57]: Good you've got rid of this 👍

**Comment by nathanrussell-moj-digital** [comment]
You have merge conflicts that need resolving; plus this doesn't build with some kind of compilation error (do you run the tests locally before committing and pushing?)

File: build.gradle.kts [hunk 9]
```diff
@@ -1,10 +1,17 @@
 plugins {
-  id("uk.gov.justice.hmpps.gradle-spring-boot") version "4.8.4"
-  kotlin("plugin.spring") version "1.8.10"
-  kotlin("plugin.jpa") version "1.8.10"
+  id("uk.gov.justice.hmpps.gradle-spring-boot") version "5.5.1"
+  kotlin("plugin.spring") version "1.9.10"
+  kotlin("plugin.jpa") version "1.8.22"
```
Comment (nathanrussell-moj-digital) [inline build.gradle.kts:4]: I suspect @petergphillips meant that all of the `kotlin("plugin.xxxxx") should be at v1.9.10 (because that is the version of kotlin you are using)
